pkg/build/admission/admission.go

```diff
@@ -0,0 +1,78 @@
+package admission
+
+import (
+	"io"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/admission"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/client"
+)
+
+type buildByStrategy struct {
+	*admission.Handler
+	client client.Interface
+}
+
+func init() {
+	admission.RegisterPlugin("BuildByStrategy", func(c kclient.Interface, config io.Reader) (admission.Interface, error) {
+		osclient := c.(client.Interface)
+		return NewBuildByStrategy(osclient), nil
+	})
+}
+
+// NewBuildByStrategy returns an admission control for builds that checks
+// on policy based on the build strategy type
+func NewBuildByStrategy(client client.Interface) admission.Interface {
+	return &buildByStrategy{
+		Handler: admission.NewHandler(admission.Create),
+		client:  client,
+	}
+}
+
+func (a *buildByStrategy) Admit(attr admission.Attributes) error {
+	if resource := attr.GetResource(); resource != "builds" {
```

Comment:
pretty sure builds/clone goes straight to the registry (via a BuildGenerator interface) without going back through admission... that may be ok if all you can do is replicate an existing build, just want to make sure we consider it